feat(playbooks): migration of centurion playbooks to github

jasonpagetas opened the pull request:
## Links


- Blocks: https://github.com/nofusscomputing/ansible_collection_centurion/issues/20
- Blocks: https://github.com/nofusscomputing/centurion_erp/pull/209


- Blocked By: https://github.com/nofusscomputing/action_project/issues/3
- Blocked By: https://github.com/nofusscomputing/ansible_collection_centurion/pull/18
- Blocked By: https://github.com/nofusscomputing/action_ansible_collection/pull/1
- Blocked By: https://github.com/nofusscomputing/action_bump/issues/5
- Blocked By: #21
- Blocked by: https://github.com/nofusscomputing/ansible_collection_centurion/issues/26


- Related: https://github.com/nofusscomputing/ansible_collection_centurion/issues/14
- Related: https://github.com/nofusscomputing/ansible_collection_centurion/issues/13
- Related: https://github.com/nofusscomputing/scratchpad/pull/8#issuecomment-2267537696

    _this PR was for the testing of CI on forked Repo's_

- Related: https://github.com/nofusscomputing/ansible_collection_centurion/issues/19

jon-nfc commented on playbooks/centurion_teams.yaml:
```diff
@@ -0,0 +1,244 @@
+- name: Centurion ERP Teams Setup
```
use two commits for these playbooks as there are two features added.

jasonpagetas replied:
Ack

jon-nfc commented on playbooks/centurion_teams.yaml:
```diff
@@ -0,0 +1,244 @@
+- name: Centurion ERP Teams Setup
+  hosts: |-
```
# ToDo - Maintainer

- CI Failure: https://github.com/nofusscomputing/ansible_collection_centurion/actions/runs/10381144478/job/28742085387

``` log

Run actions/add-to-project@244f685bbc3b7adfa8466e08b698b5577571133e
  with:
    project-url: https://github.com/orgs/nofusscomputing/projects/[3](https://github.com/nofusscomputing/ansible_collection_centurion/actions/runs/10381144478/job/28742085387#step:2:3)
Error: Input required and not supplied: github-token

```

jon-nfc replied:
issue raised, https://github.com/nofusscomputing/action_project/issues/3

- [x] Add to description

    `- Blocked By: https://github.com/nofusscomputing/action_project/issues/3`

jon-nfc replied:
- [x] Add to description

    ``` md
    - Related: https://github.com/nofusscomputing/scratchpad/pull/8#issuecomment-2267537696
    
        _this PR was for the testing of CI on forked Repo's_
    ```

jon-nfc commented on playbooks/centurion_teams.yaml:
```diff
@@ -0,0 +1,244 @@
+- name: Centurion ERP Teams Setup
```
the file names need to be fixed too

`inventory` and `teams` is sufficient.

jasonpagetas replied:
Ack, will make that change

jon-nfc commented on playbooks/centurion_teams.yaml:
```diff
@@ -0,0 +1,244 @@
+      delegate_to: localhost
+      run_once: true
+      no_log: >    # Contains a secret that logging shows
+          {{ nfc_pb_disable_log | default(true) }}
```
requires AWX import feature. see https://nofusscomputing.com/projects/ansible/playbooks/awx/#playbook-import-as-awx-ansible-automation-platform-job-template

jasonpagetas replied:
I'll need to check and potentially update this when I have access to AWX again.

jon-nfc replied:
its the same as [playbooks/itsm_inventory.yaml](https://github.com/nofusscomputing/ansible_collection_centurion/pull/17/files/0e24782c3c623c659840eff30ba5c3161ecfa47b#diff-10bead4e664e412780b944f266e65e74db655d17594c79ce8a1110274126aa54) under the `vars` key

jon-nfc commented on playbooks/itsm_inventory.yaml:
```diff
@@ -0,0 +1,171 @@
+
+    nfc_pb_awx_tower_template:
+
+      - name: "ITSM/Machine/Inventory"
```
```suggestion
      - name: "Centurion/ITAM/Inventory"
```

jasonpagetas replied:
updated in commit 26c849a6ebbad8d573d8fc18a49540d520062b3e

jon-nfc commented on playbooks/itsm_inventory.yaml:
```diff
@@ -0,0 +1,171 @@
+        ask_credential_on_launch: true
+        ask_limit_on_launch: true
+        concurrent_jobs_enabled: true
+        description: Collect inventory of host machines and publish to itsm django
```
update to reflect product name

jasonpagetas replied:
updated in commit 26c849a6ebbad8d573d8fc18a49540d520062b3e

jon-nfc commented on playbooks/itsm_inventory.yaml:
```diff
@@ -0,0 +1,171 @@
+        job_type: "run"
+        # job_tags: complete
+        labels:
+          - itsm
```
Add `centurion`

jasonpagetas replied:
updated in commit 26c849a6ebbad8d573d8fc18a49540d520062b3e

jon-nfc commented on playbooks/itsm_inventory.yaml:
```diff
@@ -0,0 +1,171 @@
+          - inventory
+        use_fact_cache: true
+        credential_types:
+          - name: 'Playbook/Inventory/Django ITSM'
```
- [ ] check this to ensure that it's using the current var names.
- [ ] update to reflect product name

current env vars are part of the inventory plugin

jasonpagetas replied:
updated in commit 26c849a6ebbad8d573d8fc18a49540d520062b3e

jon-nfc commented on playbooks/itsm_inventory.yaml:
```diff
@@ -0,0 +1,171 @@
+          package.value[0].name | regex_search("^[a-k]")
+
+
+#     - name: Inventory Software [l]
```
```suggestion
#
# This task has been commented out due to excessive memory usage (>5GB) for this playbook to complete
#
#     - name: Inventory Software [l]
```

also raise a ticket in this repo for this problem.

jasonpagetas replied:
ticket #19 raised

jon-nfc replied:
add a link to the ticket above the comment.

jasonpagetas replied:
linked

jon-nfc commented on playbooks/centurion_teams.yaml:
```diff
@@ -0,0 +1,244 @@
+- name: Centurion ERP Teams Setup
```
# To Do

when adding links directly use the link. i.e. 

- https://github.com/nofusscomputing/ansible_collection_centurion/issues/14

    this format is `- https://github.com/nofusscomputing/ansible_collection_centurion/issues/14`

using this method enables the rendered link to update.

jon-nfc commented on plugins/inventory/centurion.py:
```diff
@@ -1,5 +1,6 @@
 import json
```
why are you editing this file?

jasonpagetas replied:
I didn't edit that at all

jon-nfc replied:
if it shows in a PR, you edited it. I'm guessing you fucked the commits somewhere

jasonpagetas replied:
I'm looking at that right now...... Why the fuck is that showing up?? I only staged each of the playbooks and pushed them seperately. Didn't touch any of this other stuff

jon-nfc replied:
I'm guessing when you did a git reset you went back to the wrong commit.

I'd reset back to the dev common commit and manually re-add the commits back.

jasonpagetas replied:
Ack

jasonpagetas replied:
That looks better

jon-nfc replied:
indeed

jasonpagetas committed:
feat(playbook): Add playbook for inventory collection and publish to centurion

https://github.com/nofusscomputing/ansible_collection_centurion/pull/17 #13

jasonpagetas committed:
feat(playbook): Add playbook for teams creation and permissions

https://github.com/nofusscomputing/ansible_collection_centurion/pull/17 #14

jon-nfc commented on playbooks/inventory.yaml:
```diff
@@ -0,0 +1,172 @@
+- name: Inventory
```
# ToDo

- [x] on merge of #18 pull rebase `dev` on this branch

jon-nfc commented on playbooks/inventory.yaml:
```diff
@@ -0,0 +1,172 @@
+- name: Inventory
```
# ToDo

Update the commit footers to state `#<issue number>` they relate to and when you do the docs commit(s) do the same, this time prefix with `closes #<issue number>`. this is in additional to having the PR number before the `close` parts

jasonpagetas replied:
Updated to include issue numbers in footer

jon-nfc replied:
We use **single-line footers** and the you've put the wrong refs against both commits

jasonpagetas replied:
Dammit. How did I mess that up.

So the commit should be PR number then issue number all on one line??

jon-nfc replied:
yes. see https://www.conventionalcommits.org/en/v1.0.0/ for full spec

jasonpagetas replied:
Updated in commit [f4e6b95](https://github.com/nofusscomputing/ansible_collection_centurion/pull/17/commits/f4e6b95609c506f6bcf91c8d0ebf32df2cdb58f9) and [6b0e50a](https://github.com/nofusscomputing/ansible_collection_centurion/pull/17/commits/6b0e50a5f075c82c2708f43eaa59e66cc24d5752)

jasonpagetas commented on docs/projects/centurion_erp/teams.md:
```diff
@@ -0,0 +1,45 @@
+* **Centurion/access/teams** Creation and patching of teams and permissions
+
+On import to AWX / Ansible Automation Platform a credential type will also be created, `Playbook/teams/centurion` that can be used to supply the required secrets and Centurion host.
+
```
Initial work on the teams documentation. Don't worry about the commit message, it will be changed / removed in the future. @jon-nfc feel free to make any suggestions on how this is looking so far

jon-nfc replied:
see review comments

jon-nfc commented on docs/projects/centurion_erp/teams.md:
```diff
@@ -0,0 +1,45 @@
+# Teams creation playbook
```
requires the meta header

https://github.com/nofusscomputing/ansible_collection_centurion/blob/0c7873a51040911dd61bb84be2af13a1ee3401c2/docs/projects/ansible/collection/centurion/plugins/inventory.md?plain=1#L1-L7

jon-nfc replied:
no requirement for H1 headings

jon-nfc commented on docs/projects/centurion_erp/teams.md:
```diff
@@ -0,0 +1,45 @@
+
+The teams playbook has been created for the purpose of creating standardized teams with a defined set of permissions. With this playbook it is possible to create every team within an organization and define set teams for newly created team. The playbook can also be used to patch a team with updated permissions or notes.
+
+# Requirements
```
this should be a H2 heading.

jon-nfc replied:
FYI, two blank lines `\n` above *all* headings

jon-nfc commented on docs/projects/centurion_erp/teams.md:
```diff
@@ -0,0 +1,45 @@
+
+## Configuration
+
+The teams playbook uses variables that are located in a configuration inventory yaml file located within the host_vars folder
```
doesn't gather it from `host_vars` unless it's stored there.

I would adjust this to cover its fetched from inventory and expects to find the config under key x etc.....

jasonpagetas replied:
adjusted

jon-nfc commented on docs/projects/centurion_erp/teams.md:
```diff
@@ -0,0 +1,45 @@
+
+### Structure
+
+The expected structure of the inventory file is:
```
re-word....

structured as type x under key x etc....

jon-nfc commented on docs/projects/centurion_erp/teams.md:
```diff
@@ -0,0 +1,45 @@
+
+ > the common_permissions uses anchors to set each team_name and permissions so that they can be specified once and used multiple times. This is useful if setting up multiple organizations that require standardized teams to be created.
+
+> If only one organization is being created the common_permissions is not required.
```
this would be better suited as a `tip` admonition. see docs at https://squidfunk.github.io/mkdocs-material/reference/admonitions/#usage

jon-nfc commented on docs/projects/centurion_erp/teams.md:
```diff
@@ -0,0 +1,45 @@
+	common_permissions:
+	  name: $team-name "team name"
+	  permissions: $team-name-permissions []
+	  notes: $team-name-notes "any required notes"
```
dont use tabs

jon-nfc commented on docs/projects/centurion_erp/teams.md:
````diff
@@ -0,0 +1,45 @@
+> If only one organization is being created the common_permissions is not required.
+
+```yaml
+	teams:
````
dont use tabs

jon-nfc commented on docs/projects/centurion_erp/teams.md:
```diff
@@ -0,0 +1,45 @@
+* **Centurion/access/teams** Creation and patching of teams and permissions
+
+On import to AWX / Ansible Automation Platform a credential type will also be created, `Playbook/teams/centurion` that can be used to supply the required secrets and Centurion host.
+
```
this could be integrated into the intro paragraph.

jon-nfc commented on docs/projects/centurion_erp/teams.md:
```diff
@@ -0,0 +1,45 @@
+> If using yaml anchors replace the "team-name" with the anchor "$team-name"
+
+## AWX template
+The teams playbook includes the [AWX Feature](https://nofusscomputing.com/projects/ansible/playbooks/awx/) to import the playbook as a job template in AWX / Ansible Automation Platform.
```
dont link to the website. use relative file links. this will also require that you create the files within the docs structure. they can be empty, they just need to exist.

jon-nfc commented on docs/projects/centurion_erp/teams.md:
```diff
@@ -0,0 +1,45 @@
+* **Centurion/access/teams** Creation and patching of teams and permissions
+
+On import to AWX / Ansible Automation Platform a credential type will also be created, `Playbook/teams/centurion` that can be used to supply the required secrets and Centurion host.
+
```
for the page to show within the navigation, add it to the following and under a `Playbooks` key.

https://github.com/nofusscomputing/ansible_collection_centurion/blob/0c7873a51040911dd61bb84be2af13a1ee3401c2/mkdocs.yml#L28-L36

jasonpagetas replied:
Done

jon-nfc reviewed (changes requested):
that should be enough notes to get you started on the docs

jon-nfc commented on docs/projects/ansible/collection/centurion/playbooks/teams.md:
```diff
@@ -0,0 +1,65 @@
+about: https://gitlab.com/nofusscomputing/projects/ansible/collections/kubernetes
+---
+
+The teams playbook has been created for the purpose of creating standardised teams with a defined set of permissions. With this playbook it is possible to create every team within an organisation and define set teams for newly created team. The playbook can also be used to patch a team with updated permissions or notes.
```
elaborate or at least add that it enables the teams to be defined as part of config as code

jasonpagetas replied:
Elaborated

jon-nfc commented on docs/projects/ansible/collection/centurion/playbooks/teams.md:
```diff
@@ -0,0 +1,65 @@
+On import to AWX / Ansible Automation Platform a credential type will also be created, `Playbook/teams/centurion` that can be used to supply the required secrets and Centurion host.
+
+!!! info Centurion ERP API
+    The playbook is designed to use the Centurion ERP API to fetch, create and patch teams and permissions.
```
the playbook doesn't fetch from centurion. The inventory (centurion) plugin as part of this collection obtains what is required from centurion.

jasonpagetas replied:
Adjusted

jon-nfc commented on docs/projects/ansible/collection/centurion/playbooks/teams.md:
```diff
@@ -0,0 +1,65 @@
+
+On import to AWX / Ansible Automation Platform a credential type will also be created, `Playbook/teams/centurion` that can be used to supply the required secrets and Centurion host.
+
+!!! info Centurion ERP API
```
remove the title, it's not required. it'll default to `info`

jasonpagetas replied:
Done

jon-nfc commented on docs/projects/ansible/collection/centurion/playbooks/teams.md:
```diff
@@ -0,0 +1,65 @@
+
+## Play workflow
+
+The teams playbook gathers information regarding centurion organisations from the ansible inventory and from Centurion itself. using this information the play is designed to create new teams, patch permissions and patch notes. The workflow for the playbook is as follows
```
> from the ansible inventory and from Centurion itself

the two are one in the same. the inventory plugin within this collection builds the inventory for ansibles consumption.

jasonpagetas replied:
Think i've applied appropriate text

jon-nfc commented on docs/projects/ansible/collection/centurion/playbooks/teams.md:
```diff
@@ -0,0 +1,65 @@
+
+The teams playbook gathers information regarding centurion organisations from the ansible inventory and from Centurion itself. using this information the play is designed to create new teams, patch permissions and patch notes. The workflow for the playbook is as follows
+
+- Fetch all organisations
```
```suggestion
- Fetch all organisations from centurion
```

jasonpagetas replied:
fixed

jon-nfc commented on docs/projects/ansible/collection/centurion/playbooks/teams.md:
```diff
@@ -0,0 +1,65 @@
+The teams playbook gathers information regarding centurion organisations from the ansible inventory and from Centurion itself. using this information the play is designed to create new teams, patch permissions and patch notes. The workflow for the playbook is as follows
+
+- Fetch all organisations
+- Fetch all existing teams within each organisation
```
```suggestion
- Fetch all existing teams within each organisation from centurion
```

jasonpagetas replied:
fixed

jon-nfc commented on docs/projects/ansible/collection/centurion/playbooks/teams.md:
```diff
@@ -0,0 +1,65 @@
+- Fetch all organisations
+- Fetch all existing teams within each organisation
+- Fetch any teams to be created from inventory
+- Create any new teams that do not exist in centurion
```
```suggestion
- Create any new teams that do not exist in centurion
```

the what now????

jasonpagetas replied:
fixed

jon-nfc commented on docs/projects/ansible/collection/centurion/playbooks/teams.md:
```diff
@@ -0,0 +1,65 @@
+      teams:
+      - name: "team-name"
+        permissions: []
+        notes: "permissions must be a list"
```
tidy this up. spacing or no spacing, choose the one easier to read

jasonpagetas replied:
Done, went with no spacing

jon-nfc commented on docs/projects/ansible/collection/centurion/playbooks/teams.md:
````diff
@@ -0,0 +1,65 @@
+
+```
+
+!!! tip common teams
````
```suggestion
!!! tip "common teams"
```

titles must be in double quotes

jasonpagetas replied:
fixed

jon-nfc reviewed (changes requested):
don't forget to add to `mkdocs.yaml` the playbook files so they display within the navigation structure.

jon-nfc commented on playbooks/inventory.yaml:
```diff
@@ -0,0 +1,172 @@
+    - name: Inventory host
+      block:
+    
+      - name: Test Packages
```
```suggestion
      - name: Fetch Packages
```

jasonpagetas replied:
fixed

jon-nfc commented on playbooks/inventory.yaml:
```diff
@@ -0,0 +1,172 @@
+        when: >
+          package.value[0].name | regex_search("^[a-k]")
+
+
```
Add above this commented code the issue number where the bug is reported.

jasonpagetas replied:
fixed in commit [c482d67](https://github.com/nofusscomputing/ansible_collection_centurion/pull/17/commits/c482d67427c3545ebf636e9728e5f6353f6a708c)

jon-nfc commented on playbooks/inventory.yaml:
```diff
@@ -0,0 +1,172 @@
+        ask_credential_on_launch: true
+        ask_limit_on_launch: true
+        concurrent_jobs_enabled: true
+        description: Collect inventory of host machines and publish to centurion_erp
```
```suggestion
        description: Inventory host machines and publish to Centurion ERP
```

jasonpagetas replied:
fixed

jon-nfc commented on playbooks/inventory.yaml:
```diff
@@ -0,0 +1,172 @@
+          - itsm
+          - itam
+          - inventory
+          - centurion
```
alphanumerically order

jasonpagetas replied:
fixed

jon-nfc commented on playbooks/inventory.yaml:
```diff
@@ -0,0 +1,172 @@
+          - centurion
+        use_fact_cache: true
+        credential_types:
+          - name: 'Playbook/Inventory/Centurion'
```
```suggestion
          - name: 'Collection/No Fuss Computing/Centurion/API'
```

jasonpagetas replied:
fixed

jon-nfc commented on playbooks/inventory.yaml:
```diff
@@ -0,0 +1,172 @@
+        credential_types:
+          - name: 'Playbook/Inventory/Centurion'
+            description: |
+              Credentials for authentication to Centurion_erp
```
```suggestion
              Credentials for authentication to Centurion ERP
```

jon-nfc commented on playbooks/inventory.yaml:
```diff
@@ -0,0 +1,172 @@
+              env:
+                CENTURION_API: '{{ centurion_url }}'
+                CENTURION_TOKEN: '{{ centurion_token }}'
+                CENTURION_VALIDATE_CERTS: '{{ centurion_validate_certs | default(true) }}'
```
all files require a blank line at the end of the files `\n`.

jon-nfc commented on playbooks/teams.yaml:
```diff
@@ -0,0 +1,293 @@
+
+  nfc_pb_awx_tower_template:
+
+    - name: "Centurion/access/teams"
```
Camel case the words

jon-nfc commented on playbooks/teams.yaml:
```diff
@@ -0,0 +1,293 @@
+        - itam
+        - access
+        - permissions
+        - teams
```
alphanumerically order

jon-nfc commented on playbooks/teams.yaml:
```diff
@@ -0,0 +1,293 @@
+            env:
+              CENTURION_API: '{{ centurion_url }}'
+              CENTURION_TOKEN: '{{ centurion_token }}'
+              CENTURION_VALIDATE_CERTS: '{{ centurion_validate_certs | default(true) }}'
```
remove as it's already added in previous playbook.

jon-nfc replied:
all files to have a blank line `\n` at the end of the file.

jon-nfc commented on docs/projects/ansible/collection/centurion/playbooks/index.md:
```diff
@@ -0,0 +1,11 @@
+description: No Fuss Computings Companion Ansible Collection Centurion Playbooks.
+date: 2024-08-18
+template: project.html
+about: https://gitlab.com/nofusscomputing/projects/ansible/collections/kubernetes
```
fix linkl to be that of github

jasonpagetas replied:
fixed

jon-nfc commented on docs/projects/ansible/collection/centurion/playbooks/teams.md:
```diff
@@ -0,0 +1,65 @@
+description: No Fuss Computings Companion Ansible Collection Centurion teams Playbook.
+date: 2024-08-18
+template: project.html
+about: https://gitlab.com/nofusscomputing/projects/ansible/collections/kubernetes
```
fix link to be that of github

jasonpagetas replied:
fixed

jon-nfc commented on docs/projects/ansible/collection/centurion/playbooks/index.md:
```diff
@@ -0,0 +1,11 @@
+---
+title: Playbooks
+description: No Fuss Computings Companion Ansible Collection Centurion Playbooks.
```
the what now??

jasonpagetas replied:
Still need to adjust this

jasonpagetas replied:
Well thats what they are NFC's companion ansible collection centurion playbooks

jon-nfc replied:
```suggestion
description: Playbooks as part of No Fuss Computings Companion Ansible Collection, Centurion .
```

jasonpagetas replied:
fixed in commit [054c201](https://github.com/nofusscomputing/ansible_collection_centurion/pull/17/commits/054c201d1abd6accf7ee94843a472849075bd5e1)

jon-nfc commented on docs/projects/ansible/collection/centurion/playbooks/teams.md:
```diff
@@ -0,0 +1,65 @@
+---
+title: Teams
+description: No Fuss Computings Companion Ansible Collection Centurion teams Playbook.
```
the what now??

jasonpagetas replied:
Still need to adjust this

jasonpagetas replied:
adjusted in commit [cec6cd9](https://github.com/nofusscomputing/ansible_collection_centurion/pull/17/commits/cec6cd91a382d424c008dd89b841499e6299691a)

jon-nfc commented on docs/projects/ansible/collection/centurion/playbooks/index.md:
```diff
@@ -0,0 +1,11 @@
+---
```
can you update your dev branch so that it's up to date. hopefully that fixes the jobs that are not running

``` bash
git remote add nfc https://github.com/nofusscomputing/ansible_collection_centurion.git
git fetch --all -fpvt
git switch development
git pull nfc development --rebase
git push origin
```

jasonpagetas replied:
completed this request

jasonpagetas committed:
docs(teams): Add Documenation for teams playbook

https://github.com/nofusscomputing/ansible_collection_centurion/pull/17 #14

jasonpagetas committed:
chore(teams): Apply suggested changes

#17 #14

jasonpagetas committed:
chore(inventory): Apply suggested fixes from code review

Name changes
Description changes
linking issue to edited out task

#17 #13

jasonpagetas committed:
docs(inventory): Add documentation for inventory playbook

jasonpagetas committed:
chore(docs_teams): Update decription

remove AWX credential types as they are not created

#17 #14

jasonpagetas committed:
chore(make_docs): apply linting fixes

remove trailing spaces
fix relative links
indentation
add error to ansible-lint-ignore
Capitalise Centurion ERP

jon-nfc commented on .ansible-lint-ignore:
```diff
@@ -1 +1,5 @@
+# This playbook will only ever be run against the centurion host
+playbooks/teams.yaml run-once[task]
+# This task only saves the report on the local machine and is deleted after uploading
+playbooks/inventory.yaml risky-file-permissions
```
```suggestion
galaxy.yml galaxy[version-incorrect]

# N/A as the play runs on the ansible controller and should only run once.
playbooks/teams.yaml run-once[task]

# N/A as the play runs on the ansible controller using permissions of executing user.
playbooks/inventory.yaml risky-file-permissions

```

use spacing. **DON'T BE AFRAID OF IT!!** one line is generally enough.

jon-nfc commented on docs/projects/ansible/collection/centurion/playbooks/index.md:
```diff
@@ -0,0 +1,12 @@
+Available playbooks include:
+
+- [Inventory](./inventory.md)
+- [Teams](./teams.md)
```
nl at end of file

jasonpagetas replied:
this has been fixed

jon-nfc commented on docs/projects/ansible/collection/centurion/playbooks/inventory.md:
```diff
@@ -0,0 +1,33 @@
+about: https://github.com/nofusscomputing/ansible_collection_centurion
+---
+
+The inventory playbook has been created to inventory host machines and to publish the collected inventory to Centurion ERP. The inventory includes details of all software packages installed on the host machine as well as some details regarding the host machine such as UUID and serial number.
```
```suggestion
The inventory playbook has been created to inventory devices and to publish the collected inventory to Centurion ERP. The inventory includes details of all software packages installed on the host machine as well as some details regarding the host machine such as UUID and serial number.
```

stick to the interface terms as this aids in clarity

jasonpagetas replied:
Updated

jon-nfc commented on docs/projects/ansible/collection/centurion/playbooks/inventory.md:
```diff
@@ -0,0 +1,33 @@
+
+- Gathers host information
+- Gathers sofware information
+- POSTS an inventory report to Centurion ERP
```
```suggestion
- Uploads the inventory report to Centurion ERP
```

jasonpagetas replied:
Updated

jon-nfc commented on docs/projects/ansible/collection/centurion/playbooks/teams.md:
```diff
@@ -0,0 +1,63 @@
+
+
+!!! info 
+    The playbook is able to work with Centurion ERP directly or using the inventory pluggin that is included in this collection.
```
```suggestion
    The playbook is able to work with the inventory plugin that is included in this collection.
```

jasonpagetas replied:
Fixed

jon-nfc commented on docs/projects/ansible/collection/centurion/playbooks/teams.md:
```diff
@@ -0,0 +1,63 @@
+## Configuration
+
+The teams playbook uses variables that are gathered from inventory
+The expected structure of the inventory file is:
```
```suggestion
The teams playbook uses variables that are gathered from inventory. The expected structure of the inventory file is:
```

jasonpagetas replied:
Fixed

jon-nfc commented on docs/projects/ansible/collection/centurion/playbooks/teams.md:
```diff
@@ -0,0 +1,63 @@
+      teams:
+      - name: "team-name"
+        permissions: []
+        notes: "permissions must be a list"
```
```suggestion

centurion_erp:
  teams:
    - name: "organisation name"
      teams:
        - name: "team-name"
          permissions: []
          notes: "permissions must be a list"
```

jasonpagetas replied:
Applied change

jon-nfc commented on docs/projects/ansible/collection/centurion/playbooks/teams.md:
````diff
@@ -0,0 +1,63 @@
+    Common teams can be created by using yaml anchors. This is useful when multiple organisations require a common team and permissions to be set.
+
+    ```yaml
+    centurion_erp:
+      common_teams:
+        team_name: &team-name "team_name"
+        team_permissions: &team-name-permissions []
+        team_name_notes: &team-name-notes "team_notes"
````
add a full example that includes specifying the anchors

jasonpagetas replied:
A full example has been provided now

jon-nfc commented on docs/projects/ansible/collection/centurion/playbooks/teams.md:
````diff
@@ -0,0 +1,63 @@
+    ```
+
+!!! info "Upcoming feature"
+    With the release of Centurion ERP V1.1 an API request to fetch all permissions that can be applied will be available. 
````
this can be omitted as v1.1 will be released within hours of 1.0

jasonpagetas replied:
This has been removed

jon-nfc commented on playbooks/inventory.yaml:
```diff
@@ -0,0 +1,175 @@
+              env:
+                CENTURION_API: '{{ centurion_url }}'
+                CENTURION_TOKEN: '{{ centurion_token }}'
+                CENTURION_VALIDATE_CERTS: '{{ centurion_validate_certs | default(true) }}'
```
nl at end of file

jasonpagetas replied:
Added

jon-nfc commented on playbooks/teams.yaml:
```diff
@@ -0,0 +1,267 @@
+          - access
+          - permissions
+          - teams
+        use_fact_cache: true
```
nl at end of file

jasonpagetas replied:
Added

jon-nfc commented on docs/projects/ansible/collection/centurion/playbooks/teams.md:
```diff
@@ -0,0 +1,63 @@
+---
+title: Teams
+description: Creation and patching of teams and permissions.
```
product name is required. this is what search engines uses as part of the page description.

jasonpagetas replied:
fixed

jon-nfc commented on docs/projects/ansible/collection/centurion/playbooks/inventory.md:
```diff
@@ -0,0 +1,33 @@
+---
+title: Inventory
+description: Inventory host machines and publish to Centurion ERP
```
```suggestion
description: Inventory devices and publish to Centurion ERP
```

jasonpagetas replied:
fixed

jasonpagetas committed:
chore(): Update changes from code review

Blank lines at end of documents
rewording some docs
Spacing between ansible-lint-ignore rules
Variable names
remove uneccessry info in teams.md
changed description index
changed description teams.md

jon-nfc commented on docs/projects/ansible/collection/centurion/playbooks/teams.md:
```diff
@@ -0,0 +1,68 @@
+---
+title: Teams
+description: Creation and patching of Centurion ERP teams and permissions 
```
```suggestion
description: Ansible Playbook for Creating and patching of Centurion ERP teams, including permissions.
```

jasonpagetas replied:
fixed in commit [054c201](https://github.com/nofusscomputing/ansible_collection_centurion/pull/17/commits/054c201d1abd6accf7ee94843a472849075bd5e1)

jon-nfc commented on docs/projects/ansible/collection/centurion/playbooks/teams.md:
```diff
@@ -0,0 +1,68 @@
+
+
+!!! info 
+    The playbook is able to work with the inventory plugin that is included in this collection.
```
```suggestion
    The playbook is able to work with the [inventory plugin](../plugins/index.md) that is included in this collection.
```

jon-nfc committed:
docs(playbook_teams): add link to inventory plugin

ref: #17

jon-nfc commented on docs/projects/ansible/collection/centurion/playbooks/inventory.md:
```diff
@@ -0,0 +1,33 @@
+---
+title: Inventory
+description: Inventory devices and publish to Centurion ERP
```
```suggestion
description: Ansible Playbook to inventroy devices and publish to Centurion ERP
```

jon-nfc committed:
docs(playbook_inventory): update meta description

#17
